src/main/java/jenkins/plugins/git/OpenSSHKeyFormatImpl.java

```diff
@@ -0,0 +1,106 @@
+package jenkins.plugins.git;
+
+import hudson.FilePath;
+import io.jenkins.cli.shaded.org.apache.sshd.common.config.keys.loader.openssh.OpenSSHKeyPairResourceParser;
+import io.jenkins.cli.shaded.org.apache.sshd.common.NamedResource;
+import io.jenkins.cli.shaded.org.apache.sshd.common.config.keys.FilePasswordProvider;
+import io.jenkins.cli.shaded.org.apache.sshd.common.config.keys.writer.openssh.OpenSSHKeyPairResourceWriter;
+import io.jenkins.cli.shaded.org.apache.sshd.common.session.SessionContext;
+import io.jenkins.cli.shaded.org.apache.sshd.common.util.io.SecureByteArrayOutputStream;
```

Comment:
Shaded path to the library rather than a direct reference to the library...

Reply:
I was surprised at first as well but I think the answer lies [here](https://github.com/jenkinsci/jenkins/blob/6d80eac7e12e81fa8279f1c9daa384f55a2ce3eb/cli/pom.xml#L133), I guess what they are doing is shading the dependencies starting with org. While using library path starting as `org.apache.` the issue of NoSuchMethodExpection is encounter again.
Also check this https://github.com/jenkinsci/jenkins/blob/6d80eac7e12e81fa8279f1c9daa384f55a2ce3eb/cli/pom.xml#L74 the sshd-core v2.7.0 which is required by the binding but  is set optional. 
Will check more non this and will update on the chat.

Reply:
do not use the jenkins-cli copy, you should get this from the sshd plugin